app/src/Pages/DetailsPage/Details.tsx

```diff
@@ -38,14 +63,44 @@ function Details() {
     console.log("price history state after adding stuff in: ", priceHistory); 
   }
 
+  /*
+  Function to be called when new card rating is posted. 
+  Appends new card rating entry to the card rating array. 
+  Then, React will automatically re-render the CardDescription component
+  to include the newly computed averages of all the card ratings.
+  */
+  const handleNewCardRatingSubmission = (newCardRatingSubmission: CardRating) => {
+    console.log("card rating state before adding stuff in: ", cardRating); 
+    setCardRating(cardRating => [newCardRatingSubmission, ...cardRating]); 
+    console.log("card rating state after adding stuff in: ", cardRating); 
+
+    /**
+     * Compute the new average of the card ratings, including the newly posted card rating
+     * in your calculations. 
+     * Use reduce() iterator to get the sum of all the card ratings. Then divide by total 
+     * number of ratings.
+     */
+    if(cardRating.length <= 0) {
+      cardRatingAverage = EMPTY_CARD_RATING_MSG; 
+    }
+    else {
+      cardRatingAverage = cardRating.reduce((accumulator, currentCardRatingObj) => {
+        return accumulator + currentCardRatingObj.rating
+      }, 0) / cardRating.length; 
+    }
```

Comment:
I would convert this to a function because it is called twice, that way you can declare the function and just call it twice instead of manually writing all this code twice.

Reply:
good idea